cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -0,0 +1,172 @@
+import os
+
+import yaml
+from aws_cdk import aws_iam as iam
+from aws_cdk import aws_imagebuilder as imagebuilder
+from aws_cdk import aws_ssm as ssm
+from aws_cdk import core
+
+from common.utils import load_yaml
+from pcluster.imagebuilder import ImageBuilder
+
+
+class ImageBuilderStack(core.Stack):
+    """Create the Stack for imagebuilder."""
+
+    def __init__(self, scope: core.Construct, construct_id: str, imagebuild: ImageBuilder, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+        self.imagebuild = imagebuild
+        image_config = imagebuild.config.image_config
+        build_config = imagebuild.config.build_config
+        dev_settings_config = imagebuild.config.dev_settings_config
+
+        # TODO: use attributes from imagebuilder config instead of using these static variables.
+        core.CfnParameter(self, "EnableNvidia", type="String", default="false", description="EnableNvidia")
+        core.CfnParameter(self, "EnableDCV", type="String", default="false", description="EnableDCV")
+
+        # TODO: get archecture by ami id
+        architecture = "x86_64"
```

Comment:
instance type should be get from build_config and a validator should be implemented if the given parent image can be build using the given instance type

Reply:
yes, and this validator should be implemented in config(core) layer, I leave the validator part blank and think we should have a validation plan before implementation.